### PR TITLE
[FS-1335] Remove clients from subconversations when user is removed from main conversation

### PR DESCRIPTION
**Pull request by pcapriotti**
https://wearezeta.atlassian.net/browse/FS-1335

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: libs/wire-api/src/Wire/API/Conversation.hs [hunk 1]
```diff
@@ -232,6 +235,19 @@ instance ToSchema (Versioned 'V2 ConversationMetadata) where
           "ConversationMetadata"
           (conversationMetadataObjectSchema accessRolesSchemaV2)
 
+-- | Client list for internal API.
+data ClientList = ClientList {clClients :: [ClientId]}
```
Comment (elland) [inline libs/wire-api/src/Wire/API/Conversation.hs:239]: I want to move this somewhere, but unsure what the best place for it is. Perhaps new module with this + the handling function for the internal API call. Thoughts?

File: services/galley/src/Galley/API/Action.hs [hunk 2]
```diff
@@ -314,31 +332,16 @@ performAction tag origUser lconv action = do
       performConversationJoin origUser lconv action
     SConversationLeaveTag -> do
       let victims = [origUser]
-      E.deleteMembers (tUnqualified lcnv) (toUserList lconv victims)
-      -- update in-memory view of the conversation
-      let lconv' =
```
Comment (elland) [inline services/galley/src/Galley/API/Action.hs:319]: @pcapriotti and I took the opportunity to abstract this away 🎉

File: services/galley/test/integration/API/MLS.hs [hunk 3]
```diff
@@ -2929,3 +2937,232 @@ testLeaveRemoteSubConv = do
...
+          deleteMemberQualified (qUnqualified alice) alice qcnv
+            !!! const 200 === statusCode
+
+        -- Remove alice1 from our state as well
+        State.modify $ \mls ->
+          mls
+            { mlsMembers = Set.difference (mlsMembers mls) (Set.fromList [alice1])
+            }
+
+        -- TODO: is this the proposal from the parent or the sub? Assert both here
```
Comment (elland) [inline services/galley/test/integration/API/MLS.hs:3039]: @elland please come back and do both checks here.

**Review by stefanwire — APPROVED** [review]

**Review by pcapriotti — COMMENTED** [review]
LGTM. Can't approve formally because it's my own PR.

**Pull request merged**